ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/accordion/v1/accordion/clientlibs/site/js/accordionview.js

```diff
@@ -13,23 +13,98 @@
             description: `.${Accordion.bemBlock}__longdescription`,
             qm: `.${Accordion.bemBlock}__questionmark`,
             tooltipDiv: `.${Accordion.bemBlock}__shortdescription`,
-            label: `.${Accordion.bemBlock}__label`
+            label: `.${Accordion.bemBlock}__label`,
+            item: `.${Accordion.bemBlock}__item`
+        };
+
+        static idSuffixes = {
+            item: "-item",
+            button: "-button",
+            panel: "-panel"
+        }
+
+        static cacheKeys = {
+            buttonKey: "button",
+            panelKey: "panel",
+            itemKey: "item"
+        }
+
+        static cssClasses = {
+            button: {
+                disabled: "cmp-accordion__button--disabled",
+                expanded: "cmp-accordion__button--expanded"
+            },
+            panel: {
+                hidden: "cmp-accordion__panel--hidden",
+                expanded: "cmp-accordion__panel--expanded"
+            }
+        };
+
+        static keyCodes = {
+            ENTER: 13,
+            SPACE: 32,
+            END: 35,
+            HOME: 36,
+            ARROW_LEFT: 37,
+            ARROW_UP: 38,
+            ARROW_RIGHT: 39,
+            ARROW_DOWN: 40
+        };
+
+        static delay = 100;
+
+        static dataAttributes = {
+            item: {
+                expanded: "data-cmp-expanded"
+            }
         };
 
         constructor(params) {
             super(params);
+            const {element} = params;
+            this.#cacheElements(element);
+            if (this.#getCachedItems()) {
+                var expandedItems = this.#getExpandedItems();
+                // multiple expanded items annotated, display the last item open.
+                if (expandedItems.length > 1) {
+                    var lastExpandedItem = expandedItems[expandedItems.length - 1]
+                    this.#expandItem(lastExpandedItem);
+                    this.#collapseAllOtherItems(lastExpandedItem.id);
+                }
+                this.#refreshItems();
+                this.#bindEvents();
+            }
+            if (window.Granite && window.Granite.author && window.Granite.author.MessageChannel) {
+                /*
+                 * Editor message handling:
+                 * - subscribe to "cmp.panelcontainer" message requests sent by the editor frame
+                 * - check that the message data panel container type is correct and that the id (path) matches this specific Accordion component
+                 * - if so, route the "navigate" operation to enact a navigation of the Accordion based on index data
+                 */
+                window.CQ.CoreComponents.MESSAGE_CHANNEL = window.CQ.CoreComponents.MESSAGE_CHANNEL || new window.Granite.author.MessageChannel("cqauthor", window);
+                var _self = this;
+                window.CQ.CoreComponents.MESSAGE_CHANNEL.subscribeRequestMessage("cmp.panelcontainer", function (message) {
```

Comment:
this is a copy-paste from sites code.
Can you pls share the exact conflict we are getting from their code wrt repeatability feature (which is not really meant for authoring) that motivated us to have this code here?

Reply:
Hi Nitin, 
the copy paste was required to implement the repeatability feature. To handle repeatability we need to maintain the initial state of child items in the markup to make sure that addition of child via repeatability is done at the correct place.

This caching of the initial state needs to be intialised only after all the child in model are initialised in view.
Then , on addition of a new child, we need to bind the events to the added ,children so lots of code related to toggle, collapse and keyboard binding was getting duplicated anyway.

This is inline to how the tabsOnTop is implemented in af2